components/server/src/feature-flag/featureflag-controller.ts

```diff
@@ -24,7 +24,8 @@ export class FeatureFlagController {
     protected addSlowDatabaseFeatureFlagHandler(router: express.Router) {
         router.get("/slow-database", async (req, res) => {
             if (!User.is(req.user)) {
-                res.sendStatus(401);
+                res.setHeader("X-Gitpod-Slow-Database", "false");
```

Comment:
I think you could write it as this:

```

func getSlowDBValue(user) bool {
	if (!User.is(req.user)) {
       return false
    }

	return await getExperimentsClientForBackend().getValueAsync("slow_database", false, {
	        user: req.user,
	    });
}

value = getSlowDBValue(req.user)
res.setHeader("X-Gitpod-Slow-Database", "false");
res.sendStatus(200);
```

Reply:
Alternatively, if the try catch is needed, then use finally to perform the res & end part of the handling.

Reply:
changed to use `finally`.